### ISSUE TITLE
What changes does this make?

### ISSUE THREAD

**Issue by jkopczyn** [NLKNguyen/vim-lisp-syntax#1]
I'm looking for a relatively simple vim tool to make LISP editing easier _without_ adding half an IDE or inscrutable emacs shenanigans, like the most popular tools do. This might fit the bill, but I am not an expert in this format. Could you summarize the changes, and perhaps put them in the README?

**Comment by NLKNguyen** [comment]
Sure, 
This YAML file declares the symbols (operators/atoms) to map to highlight groups https://github.com/NLKNguyen/vim-lisp-syntax/blob/master/syntax_mapping.yml

You can review the added symbols for highlighting, and you can also modify them to include your own. It's quite declarative. You can use words of regex for matching. Then, run the generator `generate-syntax-file.js` and it will scan that YAML file to generate  `after/syntax/lisp.vim` 

You need Node.js to run the generator.

First, in the project directory, install package dependencies
```
npm install
```

Then run
```
node generate-syntax-file.js
```

FYI, the generated `lisp.vim` doesn't require the generator script or Node.js runtime.

Here is a bonus:

Since each LISP project might need different symbol highlighting, the global `lisp.vim` in your vim plugin folder might not be suitable for all projects you work on, I have an idea to have local LISP syntax per project directory so that you can optimize the syntax specifically for each LISP code base.

* LISP Project Directory /
  - .vim/
    -  after /
        - syntax /
          - lisp.vim
     - vim-lisp-syntax / [just copy these 3 files from this repos to a directory here]
       - syntax_mapping.yml
       - generate-syntax-file.js [ modify the path on [this line](https://github.com/NLKNguyen/vim-lisp-syntax/blob/master/generate-syntax-file.js#L45) to `../after/syntax/lisp.vim` ]
       - package.json 

So every time you update the syntax mapping, the generated lisp.vim is local to your project. Then, you just add ` LISP Project Directory/.vim/` to your `runtimepath` when you open Vim at this project so that only this specific LISP syntax is used when working on this LISP project.

Good luck :)
